### PR TITLE
Tls client certificate

### PR DESCRIPTION
**Pull request by DanielePalaia**
This PR add support for client certifications and the possibility to trust the certificates with a trust_certificates method.

I will add tests when https://github.com/rabbitmq/rabbitmq-stream-rust-client/pull/185 will be ready.

### CODE REVIEW

File: src/client/mod.rs [hunk 1]
```diff
@@ -638,4 +642,98 @@ impl Client {
...
+                .with_root_certificates(roots.clone())
+                .with_client_auth_cert(client_certs, client_keys.into_iter().next().unwrap())
+                .unwrap();
+        }
+
+        Ok(config)
+    }
+
+    async fn build_tls_client_configuration_trusted() -> Result<ClientConfig, ClientError> {
+        mod danger {
```
Comment (wolf4ood) [inline src/client/mod.rs:709]: I would probably rename this to `build_tls_client_configuration_untrusted`

**Review by wolf4ood — APPROVED** [review]

File: src/client/mod.rs (hunk 1 above)
Comment (DanielePalaia) [reply src/client/mod.rs]: Yeah indeed it seems like more appropriate fixed it :)

**Comment by DanielePalaia** [comment]
Let's wait for this https://github.com/rabbitmq/rabbitmq-stream-rust-client/pull/185 to be integrated in order to write a couple of tests here before integrating.

**Comment by Gsantomaggio** [comment]
FYI: the #185 has been replaced by https://github.com/rabbitmq/rabbitmq-stream-rust-client/pull/192

**Comment by Gsantomaggio** [comment]
@DanielePalaia @wolf4ood ready for review.

**Review by wolf4ood — APPROVED** [review]
LGTM

**Pull request merged**